numpy/core/src/npysort/quicksort.c.src

```diff
@@ -107,6 +111,7 @@ quicksort_@suff@(void *start, npy_intp num, void *NOT_USED)
                 *sptr++ = pi - 1;
                 pl = pi + 1;
             }
+            *psdepth++ = cdepth + 1;
```

Comment:
which is why it is increased and put on the stack, I don't see what you are getting at

maybe we should change it to recursive? the iterative conversion may not be worth the confusion
Or as a compromise make it an out the books recursive to iterative conversion without the small partition skips the depth counter trick?